functions/lib/integration/import-product.js

```diff
@@ -204,7 +214,21 @@ module.exports = ({ appSdk, storeId, auth }, tinyToken, queueEntry, appData, can
                     product.quantity = quantity >= 0 ? quantity : 0
                   }
                   console.log(`#${storeId} ${method} ${endpoint} ${product.sku} ${product.price} ${product.quantity}`)
+
                   const promise = appSdk.apiRequest(storeId, endpoint, method, product, auth)
+                    .then(async (response) => {
+                      if (appData.enable_category_import && tinyStockUpdate?.produto?.arvoreCategoria) {
+                        if (!productId) {
+                          productId = response?.response?.data && response?.response?.data._id
+                        }
+                        const { arvoreCategoria } = tinyStockUpdate?.produto
+                        if (arvoreCategoria && productId) {
```

Comment:
```suggestion
                        const { arvoreCategoria } = tinyStockUpdate.produto
                        if (productId) {
```

Três linhas pra cima já tá checando se `tinyStockUpdate?.produto?.arvoreCategoria`...